src/examples/ui/sticky-header/css/sticky-header.js

```diff
@@ -0,0 +1,162 @@
+
+	/** Resize, recalculate header height and scroll position threshold. */
+	window.addEventListener('resize', setHeaderHeight);
+
+	/** Have we scrolled below the scroll position threshold? Hide the sticky header. */
+	if (initialHide && window.pageYOffset > scrollPositionThreshold) {
+		hideStickyHeader('initial');
+	}
+
+	/** Trigger sticky header hide behavior when clicking .site-content hash links. */
+	const hashLinks = document.querySelectorAll('a[href^="#"]');
+	hashLinks.forEach(element => {
+		element.addEventListener('click', (event) => {
+			event.preventDefault();
+			const target = document.querySelector(event.target.hash);
+
+			if (!target) {
+				return;
+			}
+
+			/** Is target position above the current scroll position? Show the sticky header. */
+			if (target.offsetTop < window.pageYOffset) {
+				showStickyHeader('above');
+			}
+
+			/** Is target position below the scroll position threshold? */
+			if (target.offsetTop > window.pageYOffset) {
+				const html = document.documentElement;
+				const body = document.body;
+				const documentHeight = Math.max( body.scrollHeight, body.offsetHeight, html.clientHeight, html.scrollHeight, html.offsetHeight );
+
+				/** Is the document long enough to scroll? */
+				if (window.innerHeight < documentHeight - scrollPositionThreshold) {
+
+					/** Is the target offset below the scroll position threshold? Hide the sticky header. */
+					if (target.offsetTop > scrollPositionThreshold) {
+						hideStickyHeader('below');
+					}
+				}
+			}
+
+			window.location.hash = event.target.hash;
+		})
+	});
+
+	/** Scroll, calculate sticky header hide and show behavior. */
+	document.addEventListener('scroll', throttle(() => {
+
+		/** Wait to initialize hiding behavior. (For local links on initial load.) */
+		if (!hasScrolled) {
+			setTimeout(() => {
+				hasScrolled = true;
+			}, behaviorTimeout);
+			return;
+		}
+
+		scrollDistance = window.pageYOffset - scrollPosition;
+
+		/** Have we scrolled below the scroll position threshold? Hide the sticky header. */
+		if (window.pageYOffset > scrollPositionThreshold) {
+			scrollDistanceAbs = Math.abs(scrollDistance);
+
+			/** Have we scrolled below the scroll distance threshold? Hide or show the header. */
+			if (scrollDistanceAbs > scrollDistanceThreshold) {
+				scrollDirection = Math.sign(scrollDistance);
+
+				/** Have we changed scroll direction? */
+				if (scrollDirectionCurrent !== scrollDirection) {
+
+					/** Update scroll direction. */
+					scrollDirectionCurrent = Math.sign(scrollDirection);
+
+					/** Are we scrolling up? Show the sticky header. */
+					if (scrollDirection === -1) {
+						showStickyHeader('up ' + scrollDirection);
+					}
+
+					/** Are we scrolling down? Hide the sticky header. */
+					if (scrollDirection === 1) {
+						hideStickyHeader('down ' + scrollDirection);
+					}
+				}
+			}
+		} else {
+			/** Are we at the top of the page? Show the sticky header. */
+			showStickyHeader('top');
+		}
```

Comment:
Bit of a nitpick, is there a way we could reduce the nesting here?

Reply:
I am unsure about the best practice here but, if you see, I am setting variable values only when the previous check passes, `scrollDistanceAbs`, `scrollDirection`, `scrollDirectionCurrent`. If the check fails, nothing needs to happen. 

I suppose I could try to invert the logic for each check and return if it fails. BUT, then I still need to update the `scrollPosition` to calculate scroll distance at the bottom. So I don't think I can without sacrificing that feature. 

If you want to discuss further I am happy to talk it through

Reply:
I refactored the order of this section, leading with the _show_, `else` with the _hide_. It is still nested 4 deep, I don't think I can refactor that out